### PR TITLE
Delivery List Filter Fix

### PR DESCRIPTION
**Pull request by TurkeyFried**
This PR fixes some behavior issues with the List of Deliveries endpoint that were filtering out valid values.

Test Steps:
1. Using the Submissions endpoint, get the expanded details of a recently submitted report. e.g. GET api/waters/report/1269/history/
2. On the results, look for a Destination with an itemCount value higher than 0. Take note of the values on that Destination's organization_id, service and sentReports fields.
3. Using the previously noted values, download the list of deliveries endpoint /api/waters/org/{organization_id}.{service}/deliveries/
4. The reports in sentReports must include those seen on the Destination object for the organization you are searching for.

## Changes
- List of deliveries now checks `send` type actions instead of `receive` type actions.
- Disabled the HTTP Status filter for deliveries as they do not have a value for this.

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt [hunk 1]
```diff
@@ -166,7 +166,7 @@ abstract class HistoryDatabaseAccess(
         orgService: String?,
         since: OffsetDateTime?,
         until: OffsetDateTime?,
-        showFailed: Boolean
+        showFailed: Boolean?
```
Comment (carlosfelix2) [inline prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt:169]: Why not just set this to a default value like false?  Will be clearer I think.

**Review by carlosfelix2 — CHANGES_REQUESTED** [review]

File: prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt (hunk 1 above)
Comment (TurkeyFried) [reply prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt:169]: yeah that's a good point, updating...

Comment (TurkeyFried) [reply prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt]: give the change a look!

File: prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt [hunk 2]
```diff
@@ -178,15 +178,11 @@ abstract class HistoryDatabaseAccess(
...
-        val failedFilter: Condition = when (showFailed) {
-            true -> {
-                ACTION.HTTP_STATUS.between(200, 600)
-            }
-            false -> {
-                ACTION.HTTP_STATUS.between(200, 299)
-            }
-        }
+        if (showFailed)
+            return filter
```
Comment (carlosfelix2) [inline prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt:182]: Try not to have multiple returns.  Here you could bring up the return to the if and have an else.

**Review by carlosfelix2 — CHANGES_REQUESTED** [review]

File: prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt [hunk 3]
```diff
@@ -178,16 +178,13 @@ abstract class HistoryDatabaseAccess(
...
-                ACTION.HTTP_STATUS.between(200, 600)
-            }
-            false -> {
-                ACTION.HTTP_STATUS.between(200, 299)
-            }
+        return if (showFailed) {
+            filter
+        } else {
+            // don't show failed = need to filter by status
+            val failedFilter = ACTION.HTTP_STATUS.between(200, 299)
```
[the remark this answers is not in the document]
Comment (TurkeyFried) [reply prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt:185]: saw this right after merging. will do it on the next branch

**Pull request merged**